src/language/mongoDBService.ts

```diff
@@ -557,6 +560,87 @@ export default class MongoDBService {
     return [];
   }
 
+  // Highlight the usage of commands that only works inside interactive session.
+  provideDiagnostics(textFromEditor: string) {
+    const lines = textFromEditor.split(/\r?\n/g);
+    const diagnostics: Diagnostic[] = [];
+    const invalidInteractiveSyntaxes = [
+      { issue: 'use', fix: "use('VALUE_TO_REPLACE')", default: 'database' },
+      { issue: 'show databases', fix: 'db.getMongo().getDBs()' },
+      { issue: 'show dbs', fix: 'db.getMongo().getDBs()' },
+      { issue: 'show collections', fix: 'db.getCollectionNames()' },
+      { issue: 'show tables', fix: 'db.getCollectionNames()' },
+      {
+        issue: 'show profile',
+        fix: "db.getCollection('system.profile').find()",
+      },
+      { issue: 'show users', fix: 'db.getUsers()' },
+      { issue: 'show roles', fix: 'db.getRoles({ showBuiltinRoles: true })' },
+      { issue: 'show logs', fix: "db.adminCommand({ getLog: '*' })" },
+      {
+        issue: 'show log',
+        fix: "db.adminCommand({ getLog: 'VALUE_TO_REPLACE' })",
+        default: 'global',
+      },
+    ];
+
+    for (const [i, line] of lines.entries()) {
+      for (const item of invalidInteractiveSyntaxes) {
+        const issue = item.issue;
+        const startCharacter = line.indexOf(issue);
+        if (!line.trim().startsWith(issue)) {
+          continue;
+        }
+
+        if (issue === 'show log' && line.trim().startsWith('show logs')) {
+          continue;
+        }
+
+        let endCharacter = startCharacter + issue.length;
+        let valueToReplaceWith = item.default;
+        let fix = item.fix;
+
+        // If there is a default value, it should be used
+        // instead of VALUE_TO_REPLACE placeholder of the `fix` string.
+        if (valueToReplaceWith) {
+          const words = line.substring(startCharacter).split(' ');
+
+          // The index of the value for `use <value>` and `show log <value>`.
+          const valueIndex = issue.split(' ').length;
+
+          if (words[valueIndex]) {
+            // The `use ('database')` is a valid JS.
+            if (issue === 'use' && words[valueIndex].startsWith('(')) {
```

Comment:
`show ('logs')` is also valid JS, not just `use(...)`

Reply:
But we don't have just `show` in rules, so it won't be validated. But I see your point, we can safely remove this `issue === 'use'` validation and it won't break anything and may catch  other similar issues if we add any.